Beat Saber Utils/Gameplay/HarmonyPatches/BlahBlahGrabTheLevelData.cs

```diff
@@ -70,7 +66,7 @@ private static void __instance_didDisconnectEvent(MultiplayerLevelScenesTransiti
         }
     }
 
-    [HarmonyPatch(typeof(MissionLevelScenesTransitionSetupDataSO), nameof(MissionLevelScenesTransitionSetupDataSO.Init))]
+    [HarmonyPatch(typeof(MissionLevelScenesTransitionSetupDataSO), MethodType.Constructor)]
```

Comment:
This doesn't work; level data isn't set when playing a campaign level. It's generally a bad idea to do anything with constructors when working with Unity objects since fields won't yet be deserialized and Unity can call the constructor on an object that will never actually be used. You can target both `Init`s by removing the method name/type from the attribute and using Harmony's `TargetMethods()`:

```cs
private static IEnumerable<MethodInfo> TargetMethods()
{
    yield return AccessTools.DeclaredMethod(typeof(MissionLevelScenesTransitionSetupDataSO), nameof(MissionLevelScenesTransitionSetupDataSO.Init), new Type[] { typeof(string), typeof(IBeatmapLevelData), typeof(BeatmapKey).MakeByRefType(), typeof(BeatmapLevel), typeof(MissionObjective[]), typeof(ColorScheme), typeof(GameplayModifiers), typeof(PlayerSpecificSettings), typeof(EnvironmentsListModel), typeof(AudioClipAsyncLoader), typeof(BeatmapDataLoader), typeof(string) });
    yield return AccessTools.DeclaredMethod(typeof(MissionLevelScenesTransitionSetupDataSO), nameof(MissionLevelScenesTransitionSetupDataSO.Init), new Type[] { typeof(string), typeof(BeatmapKey).MakeByRefType(), typeof(BeatmapLevel), typeof(MissionObjective[]), typeof(ColorScheme), typeof(GameplayModifiers), typeof(PlayerSpecificSettings), typeof(EnvironmentsListModel), typeof(BeatmapLevelsModel), typeof(AudioClipAsyncLoader), typeof(BeatmapDataLoader), typeof(string) });
}
```